### PR TITLE
Moved to CMake ; removed most of Boost ; Applied Clang Format and Clang Tidy

### PR DESCRIPTION
**Pull request by Guekka**
Boost::ASIO still has to be removed. Could be replaced with https://gitlab.com/cantordust/threadpool
Please pay a particular attention to the std::chrono code, there are some places I'm not sure about.
Beware that I have not tested the new binary.

Note:  you can ignore whitespace changes in GitHub diff settings

### CODE REVIEW

File: include/generate/animation/grouptemplate.h [hunk 1]
```diff
@@ -1,110 +1,225 @@
...
-		std::vector<std::vector<std::unordered_map<std::string, bool>>>& masterOptionPicked, std::unordered_map<std::string, bool>& otherAnimType, int order, int groupOrder,
-		bool& isElement, int& elementLine, __int64& openRange, std::string& multiOption, int& elementCount, int curFunctionID, int curLine, unsigned int scopeSize);
-	void processing(std::string& line, std::string filename, int curFunctionID, int linecount, id eventid, id variableid, int groupMulti, int optionMulti = -1,
-		int animMulti = -1, std::string multiOption = "");
-	inline void newID();
-	void setZeroEvent(std::string eventname);
-	void setZeroVariable(std::string variablename);
-
+    VecStr
+    groupExistingFunctionProcess(int curFunctionID,
```
Comment (ShikyoKira) [inline include/generate/animation/grouptemplate.h:133]: why tho?

Comment (Guekka) [reply include/generate/animation/grouptemplate.h:133]: Why what?

File: include/update/dataunification.h [hunk 2]
```diff
@@ -15,106 +14,122 @@
...
+            extract.push_back(storeline[k].name);
+            newlines.clear();
+            close = false;
+        }
+        else
+        {
+            close = true;
+        }
+
+        if (close) { newlines.push_back(storeline[k].name); }
```
Comment (ShikyoKira) [inline include/update/dataunification.h:126]: why not remove the bracket tho?

Comment (Guekka) [reply include/update/dataunification.h:126]: Well, because I forgot :p

File: include/generate/animation/grouptemplate.h (hunk 1 above)
Comment (Qudix) [reply include/generate/animation/grouptemplate.h:133]: I think Skikyo is referring to the return type being on a separate line.

Comment (Guekka) [reply include/generate/animation/grouptemplate.h:133]: Oh, that. i don't know, that's clang format. I think it can be tweaked.

File: src/generate/animationdata.cpp [hunk 3]
```diff
@@ -1,867 +1,903 @@
...
+bool IDExistProcess(string change,
+                    vector<InfoDataPack>& storeline,
+                    map<string, vector<InfoDataTracker>> original,
+                    map<string, bool>& isExist,
+                    map<string, string>& exchange,
+                    vector<AnimDataPack>& animDataPack,
+                    map<string, vector<int>>& codeTracker,
+                    map<string, bool>& loopCheck);
+void BehaviorListProcess(
+    AnimDataProject& storeline, int& startline, VecStr& animdatafile, string project, string modcode);
```
Comment (ShikyoKira) [inline src/generate/animationdata.cpp:30]: shouldn't it be `void BehaviorListProcess(AnimDataProject& storeline`?

Comment (Qudix) [reply src/generate/animationdata.cpp:30]: Yes, it's just misplaced

File: include/generate/animation/grouptemplate.h (hunk 1 above)
Comment (ShikyoKira) [reply include/generate/animation/grouptemplate.h:133]: yea i meant the return type being on a separate line. I have no problem with using Capital letter VecStr

**Comment by Guekka** [comment]
This PR is not ready yet. I tried the binary and found some bugs, I'm working on it.

**Comment by ShikyoKira** [comment]
alright

File: src/generate/animation/singletemplate.cpp [hunk 4]
```diff
@@ -7,2670 +9,3145 @@
...
+             OptionList& optionlist)
 {
-	if (cond.length() > 0)
-	{
-		condition = make_shared<condt>(cond, format, behaviorFile, cond, multiOption, numline, isGroup, isMaster, optionlist);
-	}
-	else
-	{
-		condition = make_shared<condt>(format, format, behaviorFile, cond, multiOption, numline, isGroup, isMaster, optionlist);
-	}
```
Comment (ShikyoKira) [inline src/generate/animation/singletemplate.cpp:1588]: why is this part removed?

Comment (Guekka) [reply src/generate/animation/singletemplate.cpp:1588]: Good find. It took me a few minutes to remember. Actually, it was not removed, it was moved to templateprocessing.cpp
It wouldn't compile without it, I still do not understand. So I did not bother and I moved it

File: src/generate/animation/singletemplate.cpp [hunk 5]
```diff
@@ -7,2670 +9,3145 @@
...
+
+        pos = line.find("$MC$");
+
+        // multi choice selection
+        if (pos != NOT_FOUND)
+        {
+            vector<multichoice> m_conditions;
+            process.hasMC[numline] = true;
+
+            std::regex reg(std::regex("[\\s]+<!-- (.+?) -->[\\s]*?"));
```
[the remark this answers is not in the document]
Comment (Guekka) [reply src/generate/animation/singletemplate.cpp:542]: You're right, that's a typo

**Comment by Guekka** [comment]
Please check #197 instead
